src/dune_pkg/lock_dir.ml

```diff
@@ -555,15 +597,24 @@ module Write_disk = struct
         Format.asprintf "%a" Pp.to_fmt pp |> Io.write_file path;
         Package_name.Map.iteri files ~f:(fun package_name files ->
           let files_dir =
-            Pkg.files_dir package_name ~lock_dir:lock_dir_path_src |> Path.source
+            Path.relative lock_dir_path (Package_name.to_string package_name ^ ".files")
           in
           Path.mkdir_p files_dir;
           List.iter files ~f:(fun { File_entry.original; local_file } ->
             let dst = Path.append_local files_dir local_file in
             Path.mkdir_p (Path.parent_exn dst);
             match original with
             | Path src -> Io.copy_file ~src ~dst ()
-            | Content content -> Io.write_file dst content)))
+            | Content content -> Io.write_file dst content)));
+      safely_rename_lock_dir_thunk ~dst:lock_dir_path_external lock_dir_path ();
+      remove_hidden_dir_if_exists ()
+    in
+    fun () ->
+      Temp.with_temp_dir
+        ~parent_dir:(Path.source Path.Source.root)
```

Comment:
I think the directory should be derived from the directory where the lock directory is being placed rather than hardcoded to be the source root.

Reply:
I have updated the code following your suggestion. It puts the temporary directory in the lock dir parent. In most case it will be `.` as the lock dir is `./dune.lock`